tdh/resource_data_plane.go

```diff
@@ -166,18 +181,15 @@ func (r *dataPlaneResource) Create(ctx context.Context, req resource.CreateReque
 	dataPlaneRequest := infra_connector.DataPlaneCreateRequest{
 		AccountId:             plan.AccountId.ValueString(),
 		DataplaneName:         plan.Name.ValueString(),
-		K8sClusterName:        plan.K8sClusterName.ValueString(),
 		DataplaneType:         "regular",
-		BackupStorageClass:    plan.BackupStorageClass.ValueString(),
 		ManagedDns:            true,
-		DataPlaneReleaseId:    plan.DataPlaneReleaseId.ValueString(),
-		Shared:                plan.Shared.ValueBool(),
-		AutoUpgrade:           plan.AutoUpgrade.ValueBool(),
-		CpBootstrappedCluster: plan.CpBootstrappedCluster.ValueBool(),
-		ConfigureCoreDns:      plan.ConfigureCoreDns.ValueBool(),
+		Shared:                true,
+		AutoUpgrade:           false,
+		CpBootstrappedCluster: false,
+		ConfigureCoreDns:      true,
 	}
 
-	if !plan.OrgId.IsNull() {
+	if !plan.OrgId.IsNull() && plan.ProviderName.ValueString() != "tas" {
```

Comment:
add a single validation method for normal & conditional inputs. like what if user puts shared=true AND give org_id.. or user doesn't pass field required for TAS or non-TAS

Reply:
this is not a validation method. Here we are setting request body fields based on the condition.